tests/test.rs

```diff
@@ -1012,13 +1014,29 @@ fn test_parse_string() {
             &[b'"', b'\\', b'n', 159, 146, 150, b'"'],
             "invalid unicode code point at line 1 column 7",
         ),
+        (
+            &[b'"', b'\\', b'u', 250, 48, 51, 48, b'"'],
+            "invalid escape at line 1 column 4",
+        ),
+        (
+            &[b'"', b'\\', b'u', 48, 250, 51, 48, b'"'],
+            "invalid escape at line 1 column 5",
+        ),
+        (
+            &[b'"', b'\\', b'u', 48, 48, 250, 48, b'"'],
+            "invalid escape at line 1 column 6",
+        ),
+        (
+            &[b'"', b'\\', b'u', 48, 48, 51, 250, b'"'],
+            "invalid escape at line 1 column 7",
+        ),
         (
             &[b'"', b'\n', b'"'],
-            "control character (\\u0000-\\u001F) found while parsing a string at line 1 column 1",
+            "control character (\\u0000-\\u001F) found while parsing a string at line 2 column 0",
```

Comment:
I'm not sure `line 2 column 0` is a valid position or not. I left it with `line 2 column 0` because
 - `line 1 column 1` points to `"` which is wrong.
 - recovering a column information from `IoRead` seems quite expensive